src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1062,6 +1062,43 @@ private bool ImportIntrinsicCall(MethodDesc method)
 
             switch (method.Name)
             {
+                case "InitializeArray":
+                    if (metadataType.Namespace == "System.Runtime.CompilerServices" && metadataType.Name == "RuntimeHelpers")
+                    {
+                        var fieldSlot = (LdTokenEntry<FieldDesc>)_stack.Pop();
+                        var arraySlot = _stack.Pop();
+
+                        var node = (BlobNode)_compilation.GetFieldRvaData(fieldSlot.LdToken);
+
+                        LLVMValueRef src = LoadAddressOfSymbolNode(node);
+                        _dependencies.Add(node);
+                        int srcLength = node.GetData(_compilation.NodeFactory, false).Data.Length;
+
+                        LLVMValueRef arrayObjPtr = arraySlot.ValueAsType(LLVM.PointerType(LLVM.Int8Type(), 0), _builder);
+
+                        var argsType = new LLVMTypeRef[]
+                        {
+                            LLVM.PointerType(LLVM.Int8Type(), 0),
+                            LLVM.PointerType(LLVM.Int8Type(), 0),
+                            LLVM.Int32Type(),
+                            LLVM.Int32Type(),
+                            LLVM.Int1Type()
+                        };
+                        var memcpyFunction = GetOrCreateLLVMFunction("llvm.memcpy.p0i8.p0i8.i32", LLVM.FunctionType(LLVM.VoidType(), argsType, false));
+
+                        var args = new LLVMValueRef[]
+                        {
+                            LLVM.BuildGEP(_builder, arrayObjPtr, new LLVMValueRef[] { ArrayBaseSize() }, String.Empty),
```

Comment:
This doesn't include space for multidimensional array lengths. If you don't want to handle those, it's ok, but please add a check and throw NotImplementedException.

Reply:
I do want to handle those, however I cannot test on multidimensional arrays; see #5421.